### PR TITLE
Allow user to set priority for a task

### PR DESCRIPTION
**Pull request by xiaoral2**
[(task:18335)](https://staging.teamyapp.com/teams/107/tasks/18335)

### CODE REVIEW

File: core/api/gql/schema.graphql [hunk 1]
```diff
@@ -431,6 +432,7 @@ input UpdateTaskInput {
 	owningTeamId: ID!
 	effort: Duration
 	dueAt: Time
+	priority: Priority
```
Comment (naturetime10) [inline core/api/gql/schema.graphql:434]: Move priority above dueAt

**Review by naturetime10 — CHANGES_REQUESTED** [review]

**Comment by naturetime10** [comment]
Please remove "backend" from PR title.

**Comment by naturetime10** [comment]
Can we also add this field to all gRPC APIs related to task?

**Comment by naturetime10** [comment]
You need to install protoc, golang grpc plugin, and then run ./scripts/proto_gen.js to generate updated client and server code for gRPC in Go.

**Review by naturetime10 — APPROVED** [review]

File: core/api/proto/task.proto [hunk 2]
```diff
@@ -29,19 +29,28 @@ enum TaskStatus {
...
-	uint64 commentThreadId = 12;
+	optional Priority priority = 6;
+	TaskStatus status = 7;
+	google.protobuf.Timestamp createdAt = 8;
+	optional google.protobuf.Timestamp updatedAt = 9;
+	optional uint64 ownerUserId = 10;
+	uint64 owningTeamId = 11;
+	uint64 creatorUserId = 12;
+	uint64 commentThreadId = 13;
+
```
Comment (duanxianbo) [inline core/api/proto/task.proto]:
```suggestion
```

**Pull request merged**